src/lib.rs

```diff
@@ -126,28 +126,31 @@ mod tests;
 pub use self::{
 	meta_type::MetaType,
 	registry::{IntoCompact, Registry},
-	type_def::*,
-	type_id::*,
+	ty::*,
 };
 
 #[cfg(feature = "derive")]
-pub use type_metadata_derive::{Metadata, TypeDef, TypeId};
+pub use type_metadata_derive::Metadata;
 
-/// A super trait that shall be implemented by all types implementing
-/// `HasTypeId` and `HasTypedef` in order to more easily manage them.
+/// A super trait that shall be implemented by all types implementing `TypeInfo`
 ///
-/// This trait is automatically implemented for all `'static` type that
-/// also implement `HasTypeId` and `HasTypeDef`. Users of this library should
-/// use this trait directly instead of using the more fine grained
-/// `HasTypeId` and `HasTypeDef` traits.
-pub trait Metadata: HasTypeId + HasTypeDef {
+/// This trait is automatically implemented for all `'static` types that also
+/// implement `TypeInfo`. Users of this library should use this trait directly
+/// instead of using the `TypeInfo` trait.
+pub trait Metadata: TypeInfo {
 	/// Returns the runtime bridge to the types compile-time type information.
 	fn meta_type() -> MetaType;
 }
```

Comment:
The only reason why we have this trait is to have the auto-impl defined below. So there is no longer the need for a derive macro for this since types should simply implement `TypeInfo` and have a `Metadata` impl right away. Correct?